py_mmd_tools/odajson_to_mmd.py

```diff
@@ -0,0 +1,306 @@
+            'file '
+            'aborted.')
+        return None
+
+    # Merge default and dataset specific elements
+    # (default is overwritten if it's also defined at dataset level)
+    elements_out = copy.deepcopy(default_elements)
+    _merge_dicts(elements_out, dataset_elements)
+
+    # Rename elements to follow mmd vocabulary for data production status
+    choices = {'open': 'In Work', 'future': 'Planned', 'closed': 'Complete'}
+    elements_out['dataset_production_status'] = choices.get(elements_out['production_status'], None)
+    if elements_out['dataset_production_status'] is None:
+        logger.warning(
+            'The value for key %s (%s) is not a valid choice. Correct choices are: '
+            'open/future/closed. This element '
+            'is required, so the creation of MMD file is aborted.' % (
+                'production_status', elements_out['production_status']))
+        return None
+
+    # Remove operational status if empty
+    if elements_out['operational_status'] == '':
+        elements_out.pop('operational_status')
+    else:
+        elements_out['operational_status'] = elements_out['operational_status'].title()
+
+    # Remove end date for ongoing data
+    elements_out['temporal_extent'].pop('end_date', None)
+
+    # Create dataset specific title and abstract if possible
+    try:
+        elements_out['title_full'] = elements_out[
+                                         'keywords_cf'] + ' observations from weather station ' + \
+                                     elements_out['station_name'] + ' (station ID ' + elements_out[
+                                         'station_id'] + ').'
+        elements_out['abstract_full'] = 'Timeseries of ' + elements_out[
+            'keywords_cf'] + ' observations from the Norwegian weather station ' + elements_out[
+                                            'station_name'] + ' (station ID ' + elements_out[
+                                            'station_id'] + '). The observations have been ' \
+                                                            'through the data ' \
+                                                            'collection system of the Norwegian ' \
+                                                            'Meteorological ' \
+                                                            'institute which includes a number of ' \
+                                                            'automated and ' \
+                                                            'manual quality control routines. The ' \
+                                                            'number of available ' \
+                                                            'quality control routines is element ' \
+                                                            'dependent. '
+    except KeyError:
+        elements_out['title_full'] = elements_out['title']
+        elements_out['abstract_full'] = elements_out['abstract']
```

Comment:
```suggestion
try:
    elements_out['abstract_full'] = elements_out['abstract']
except KeyError:
    elements_out['abstract_full'] = 'Timeseries of ' + elements_out[ ...
```

Reply:
Regarding coverage, it's very confusing. When I run the Vagrant up or nosetests locally, I get 100% coverage for the odajson_to_mmd.py but then if I push to github, codecov gives me a different result. 

See for example (only 99% as I added a new function right now and have not updated tests yet):

    default: Attaching to vagrant_tests_1
    default: tests_1  | ....................................
    default: tests_1  | Name                             Stmts   Miss  Cover   Missing
    default: tests_1  | --------------------------------------------------------------
    default: tests_1  | py_mmd_tools/__init__.py             0      0   100%
    default: tests_1  | py_mmd_tools/log_util.py            30      3    90%   22-24
    default: tests_1  | py_mmd_tools/odajson_to_mmd.py     121      1    99%   34
    default: tests_1  | py_mmd_tools/xml_utils.py           46      4    91%   84, 87, 92-93
    default: tests_1  | --------------------------------------------------------------
    default: tests_1  | TOTAL                              197      8    96%
    default: tests_1  | ----------------------------------------------------------------------
    default: tests_1  | Ran 36 tests in 62.396s

Have you noticed that as well?

I'll first update the tests with mock and then looked into why I get different coverages with codecov.

Reply:
yes, I have.. it's a bit strange...